### PR TITLE
Allow FileSystem.exists to follow symlinks or not

### PR DESCRIPTION
**Pull request by hartbit**

### CODE REVIEW

**Comment by hartbit** [comment]
@swift-ci please smoke test

File: Tests/BasicTests/FileSystemTests.swift [hunk 1]
```diff
@@ -71,6 +71,37 @@ class FileSystemTests: XCTestCase {
...
 
+    func testLocalExistsSymlink() throws {
+        mktmpdir { path in
+            let fs = Basic.localFileSystem
+
+            let source = path.appending(component: "source")
+            let target = path.appending(component: "target")
+            try fs.writeFileContents(target, bytes: "source")
+
+            // source and target exist
```
Comment (aciidb0mb3r) [inline Tests/BasicTests/FileSystemTests.swift]: Nit: Capitalize and punctuation.

```
// Source nad target exists.
```

File: Tests/BasicTests/FileSystemTests.swift [hunk 2]
```diff
@@ -71,6 +71,37 @@ class FileSystemTests: XCTestCase {
...
+
+            try createSymlink(source, pointingAt: target)
+            XCTAssertEqual(fs.exists(source), true)
+            XCTAssertEqual(fs.exists(source, followSymlink: true), true)
+            XCTAssertEqual(fs.exists(source, followSymlink: false), true)
+
+            // source only exists
```
Comment (aciidb0mb3r) [inline Tests/BasicTests/FileSystemTests.swift]: Nit: Capitalize and punctuation.

File: Tests/BasicTests/FileSystemTests.swift [hunk 3]
```diff
@@ -71,6 +71,37 @@ class FileSystemTests: XCTestCase {
...
+
+            try fs.removeFileTree(target)
+            XCTAssertEqual(fs.exists(source), false)
+            XCTAssertEqual(fs.exists(source, followSymlink: true), false)
+            XCTAssertEqual(fs.exists(source, followSymlink: false), true)
+            
+            // none exist
```
Comment (aciidb0mb3r) [inline Tests/BasicTests/FileSystemTests.swift]: Nit: Capitalize and punctuation.

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Pull request merged**